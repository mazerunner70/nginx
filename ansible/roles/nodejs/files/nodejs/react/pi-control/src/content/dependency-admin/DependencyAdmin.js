import React from 'react';
import { textFilter } from 'react-bootstrap-table2-filter';

import 'App.css';

import DataGrid from '../../components/datagrid/DataGrid';
import { initialiseDependencyAdmin, updateStore } from './DependencyAdminActions';
import { store } from '../../store';



export default class DependencyAdmin extends React.Component {
  constructor(props) {
    super(props);
    this.submitValues = this.submitValues.bind(this);
    store.dispatch(initialiseDependencyAdmin( [] ));
  }

  columnData = {
    columns: [{
      dataField: 'name',
      text: 'Name',
    },
    {
      dataField: 'currVer',
      text: 'Current Version',
    }, {
      dataField: 'lastCheck',
      text: 'Last Checked',
      sort: true,
    }]
  }
// [{ name: 'dummy',
//   currVer: 2.4,
//   lastCheck: 2018-01-01T01:20:30.000Z }]

  submitValues(values) {
    console.log('030','got here');
    console.log(values);
    store.dispatch(updateStore(values));
  }


  render() {
    console.log('==',this.columnData);
    console.log(store.getState());
    return (
      <div className='w3-row w3-padding-64'>
        <div className='w3-container'>
          <h1 className='w3-text-teal'>Dependency Admin</h1>
          <DataGrid 
            columnData={this.columnData} 
            content={store.getState().dependencies || []}
            processSubmitedValues={this.submitValues} />
        </div>
      </div>
    );
  }

}
          // <DataGrid columnData={this.columnData} content={store.getState().dependencies}/>
