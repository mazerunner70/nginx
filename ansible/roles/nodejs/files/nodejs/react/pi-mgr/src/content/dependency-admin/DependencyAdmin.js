import React from 'react';
import styled from 'styled-components';

import DataGrid from '../../components/datagrid/DataGrid';
import { initialiseDependencyAdmin, dataSubmitted } from './DependencyAdminActions';
import { store } from '../../store';

const MainPaneDiv = styled.div `
  padding-top:64px!important;
  padding-bottom:64px!important
  &:before &:after {
    content:"";
    display:table;
    clear:both;
  }
`;

const InnerPaneDiv = styled.div `
  padding:0.01em 16px
  &:before &:after {
    content:"";
    display:table;
    clear:both;
`;

const Header = styled.h1 `
    color:#009688!important
`;


export default class DependencyAdmin extends React.Component {
  constructor(props) {
    super(props);
    this.submitValues = this.submitValues.bind(this);
    store.dispatch(initialiseDependencyAdmin( [] ));
  }

  columnData = {
    columns: [{
      dataField: 'name',
      text: 'Name',
      prompt: 'Registered name of dependency',
      sort: true,
    },
    {
      dataField: 'currVer',
      text: 'Current Version',
      prompt: 'Version last found'
    }, {
      dataField: 'lastCheck',
      text: 'Last Checked',
      prompt: 'Date checked'
    }]
  }

// [{ name: 'dummy',
//   currVer: 2.4,
//   lastCheck: 2018-01-01T01:20:30.000Z }]

  submitValues(values) {
    console.log('030','got here');
    console.log(values);
    store.dispatch(dataSubmitted(values));
  }


  render() {
    console.log('==',this.columnData);
    console.log(store.getState());
    return (
      <MainPaneDiv>
        <InnerPaneDiv>
          <Header>Dependency Admin</Header>
          <DataGrid 
            columnData={this.columnData} 
            content={store.getState().dependencies || []}
            processSubmitedValues={this.submitValues} />
        </InnerPaneDiv>
      </MainPaneDiv>
    );
  }

}
          // <DataGrid columnData={this.columnData} content={store.getState().dependencies}/>
