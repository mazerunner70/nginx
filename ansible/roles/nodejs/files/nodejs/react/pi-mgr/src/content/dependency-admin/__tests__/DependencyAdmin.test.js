import React from 'react';
import renderer from 'react-test-renderer';

import DependencyAdmin from '../DependencyAdmin';

test('Test rendering of Dependency Admin', () => {
  const component = renderer.create(
    <DependencyAdmin/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
