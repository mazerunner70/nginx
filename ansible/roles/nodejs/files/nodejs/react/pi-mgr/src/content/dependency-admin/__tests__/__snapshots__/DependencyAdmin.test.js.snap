// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rendering of Dependency Admin 1`] = `
<div
  className="sc-VigVT hnExUH"
>
  <div
    className="sc-jTzLTM dItSWa"
  >
    <h1
      className="sc-fjdhpX bpefbq"
    >
      Dependency Admin
    </h1>
    <div
      className="sc-bdVaJa bEjzhO"
    >
      <div
        className="sc-bwzfXH sUFFa"
      >
        <div
          className="sc-bxivhb dtyIAg"
        >
          <div
            className="react-bootstrap-table"
          >
            <table
              className="table table-bordered"
              id={undefined}
            >
              <thead>
                <tr
                  className={undefined}
                >
                  <th
                    className="sortable"
                    onClick={[Function]}
                  >
                    Name
                    <span
                      className="order"
                    >
                      <span
                        className="dropdown"
                      >
                        <span
                          className="caret"
                        />
                      </span>
                      <span
                        className="dropup"
                      >
                        <span
                          className="caret"
                        />
                      </span>
                    </span>
                  </th>
                  <th>
                    Current Version
                  </th>
                  <th>
                    Last Checked
                  </th>
                </tr>
              </thead>
            </table>
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH sUFFa"
      >
        <div
          className="sc-ifAKCX fwEYjo"
        >
          <span
            className="sc-EHOje fZEFZv"
          >
            Select row to edit
          </span>
          <form
            onSubmit={[Function]}
          >
            <ul
              className="sc-bZQynM hozfSL"
            >
              <li
                className="sc-gzVnrw ldtKPt"
              >
                <label
                  className="sc-htoDjs hvdFIO"
                  htmlFor="name"
                >
                  Name
                </label>
                <input
                  className="sc-dnqmqq kIdgNu"
                  maxLength="100"
                  name="name"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <span
                  className="sc-iwsKbI hAMedX"
                >
                  Registered name of dependency
                </span>
              </li>
              <li
                className="sc-gzVnrw ldtKPt"
              >
                <label
                  className="sc-htoDjs hvdFIO"
                  htmlFor="currVer"
                >
                  Current Version
                </label>
                <input
                  className="sc-dnqmqq kIdgNu"
                  maxLength="100"
                  name="currVer"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <span
                  className="sc-iwsKbI hAMedX"
                >
                  Version last found
                </span>
              </li>
              <li
                className="sc-gzVnrw ldtKPt"
              >
                <label
                  className="sc-htoDjs hvdFIO"
                  htmlFor="lastCheck"
                >
                  Last Checked
                </label>
                <input
                  className="sc-dnqmqq kIdgNu"
                  maxLength="100"
                  name="lastCheck"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <span
                  className="sc-iwsKbI hAMedX"
                >
                  Date checked
                </span>
              </li>
              <li
                className="sc-gzVnrw ldtKPt"
              >
                <input
                  className="sc-gZMcBi jnTvlg"
                  type="submit"
                  value="Submit"
                />
                <button
                  className="sc-gqjmRU kfKnXj"
                  onClick={[Function]}
                  type="reset"
                >
                  Clear
                </button>
              </li>
            </ul>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
